WcaOnRails/app/jobs/singleton_application_job.rb

```diff
@@ -0,0 +1,9 @@
+# frozen_string_literal: true
+
+class SingletonApplicationJob < ApplicationJob
+  before_enqueue do |job|
+    # Abort if job of the kind is already enqueued.
+    already_enqueued = Delayed::Job.exists?(["handler LIKE ?", "%#{job.class.name}%"])
```

Comment:
Haha, clever! Maybe slightly better to check for the yaml key as well?

```SQL
["handler LIKE ?", "%job_class: #{job.class.name}%"]
```

Reply:
Definitely cleaner and more reliable, thanks!